app/scripts/background.js

```diff
@@ -901,20 +921,17 @@ export function setupController(
 
   setupSentryGetStateGlobal(controller);
 
-  const isClientOpenStatus = () => {
-    return (
-      openPopupCount > 0 ||
-      Boolean(Object.keys(openMetamaskTabsIDs).length) ||
-      notificationIsOpen
-    );
-  };
-
   const onCloseEnvironmentInstances = (isClientOpen, environmentType) => {
     // if all instances of metamask are closed we call a method on the controller to stop gasFeeController polling
     if (isClientOpen === false) {
       controller.onClientClosed();
-      // otherwise we want to only remove the polling tokens for the environment type that has closed
+      // Set extension to idle when all UI instances are closed
+      extensionUpdateManager.setIdleState(true);
+      // Check for updates when UI is closed
+      extensionUpdateManager.applyPendingUpdateIfNeeded();
```

Comment:
We're calling this redundantly twice in this file, because this gets called by `setIdleState` as well. Can we remove both calls to `applyPendingUpdateIfNeeded` in this file, and make it private? Or would you prefer to keep this, and remove the call from inside `setIdleState`? The former seems simpler to me.

Reply:
Yeah, this is called duplicated indeed. removed 9f7c7f3f0b6fe0b343ae1cbce6d59b17ea084e14